registry/api/errcode/errors.go

```diff
@@ -97,25 +101,28 @@ type Error struct {
 
 var _ error = Error{}
 
-// ErrorCode returns the ID/Value of this Error
+// ErrorCode returns the ID/Value of this Error.
 func (e Error) ErrorCode() ErrorCode {
 	return e.Code
 }
 
 // Error returns a human readable representation of the error.
 func (e Error) Error() string {
-	return fmt.Sprintf("%s: %s",
```

Comment:
Exactly!  Let the code that defines the error set the error text to whatever they want and that exact text should be displayed. It shouldn't be prepended with something that they don't want.

The ID of the error, right now, isn't useful to the end user so I don't see the value in showing it. They can't do anything with that information.  At some point when we i18n our messages, or have a catalog of errors messages where that ID could be used for quick look-up, then it makes more sense but right now we don't have those situations and I'd prefer to wait until we have a real use for them instead of forcing it on all Docker error messages right now. And at that point in time I think we need to get agreement from the broader community that we're going to add these IDs to all error message's text.